File: app/src/main/java/com/gustafson/ctcandroid_new/SplashScreenActivity.java
package com.gustafson.ctcandroid_new;

import android.os.Bundle;
import android.support.constraint.motion.MotionLayout;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;


public class SplashScreenActivity extends FragmentActivity {
    MotionLayout motionLayout;

    @Override
    protected void onCreate(Bundle savedInstances){
        super.onCreate(savedInstances);
        setContentView(R.layout.splash_screen_view_container);
        FragmentManager fm = getSupportFragmentManager();
        FragmentTransaction ft = fm.beginTransaction();
        SplashScreenFragment ssf = new SplashScreenFragment();
        ft.add(R.id.fragment_container, ssf);
        ft.commit();

    }

}

File: app/src/main/java/com/example/codethatcaresandroid/SplashScreenActivity.kt
package com.example.codethatcaresandroid

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.os.PersistableBundle
import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction

class SplashScreenActivity : FragmentActivity(){
    var prefs: SharedPreferences? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        prefs = applicationContext?.getSharedPreferences("com.codethatcares.ctcandroid", Context.MODE_PRIVATE)
        if (!prefs!!.getBoolean("splash", true)){
            val intent: Intent = Intent(applicationContext, MainActivity::class.java)
            startActivity(intent)
        }
        setContentView(R.layout.splash_screen_view_container)
        val fm: FragmentManager = supportFragmentManager
        val ft: FragmentTransaction = fm.beginTransaction()
        val ssf: SplashScreenFragment = SplashScreenFragment()
        ft.add(R.id.fragment_container, ssf)
        ft.commit()
    }
}
